src/http.js

```diff
@@ -0,0 +1,33 @@
+const shared = require('./utils/shared');
+
+const preload = (content, resourcePath, skyPagesConfig) => {
+  if (!resourcePath.match(/app-extras\.module\.ts$/)) {
+    return content;
+  }
+
+  const useAuth = skyPagesConfig.skyux.auth;
+
+  function injectHttpService() {
+    if (useAuth) {
+      return `import { SkyAuthHttp } from '@blackbaud/skyux-builder/runtime';`;
+    } else {
+      return `import { Http } from '@angular/http';`;
+    }
+  }
+
+  const modulePath = shared.getModulePath(resourcePath);
+
+  content = `
+${injectHttpService()}
+import { StacheHttpService } from '${modulePath}';
+
+export const STACHE_HTTP_PROVIDERS: any[] = [{
+  provide: StacheHttpService,
+  useExisting: ${useAuth ? `SkyAuthHttp` : `Http`}
```

Comment:
I'm curious about using a function to get one value versus ternary in another.  Any reason not to do something like...
```
const shared = require('./utils/shared');

const preload = (content, resourcePath, skyPagesConfig) => {
  if (!resourcePath.match(/app-extras\.module\.ts$/)) {
    return content;
  }

  const modulePath = shared.getModulePath(resourcePath);
  let httpName;
  let httpPath;

  if (skyPagesConfig.skyux.auth) {
    httpName = 'SkyAuthHttp';
    httpPath = '@blackbaud/skyux-builder/runtime';
  } else {
    httpName = 'Http';
    httpPath = '@angular/http';
  }

  content = `
import { ${httpName} } from '${httpPath}';
import { StacheHttpService } from '${modulePath}';

export const STACHE_HTTP_PROVIDERS: any[] = [{
  provide: StacheHttpService,
  useExisting: ${httpName}
}];
${content}`;

  return shared.addToProviders(content, 'STACHE_HTTP_PROVIDERS');
};

module.exports = { preload };
```